ocean_lib/data_provider/data_service_provider.py

```diff
@@ -259,7 +263,12 @@ def download(
     def sign_message(wallet: Wallet, msg: str) -> Tuple[str, str]:
         nonce = str(datetime.utcnow().timestamp())
         print(f"signing message with nonce {nonce}: {msg}, account={wallet.address}")
-        return nonce, sign_hash(encode_defunct(text=f"{msg}{nonce}"), wallet)
+        # keys_pk = keys.PrivateKey(Web3.toBytes(text=wallet.private_key))
+        message_hash = Web3.solidityKeccak(
+            ["bytes"],
+            [Web3.toHex(Web3.toBytes(text=f"{msg}{nonce}"))],
```

Comment:
You don't need both `toHex` and `toBytes` here. Please simplify to just use one or the other. I recommend using `Web3.toBytes(text=message)` for consistency with all the other PRs I left this same comment on :sweat_smile: 
 
```python
>>> from web3.main import Web3
>>> msg = "hello"
>>> Web3.solidityKeccak(["bytes"], [Web3.toHex(Web3.toBytes(text=msg))])
HexBytes('0x1c8aff950685c2ed4bc3174f3472287b56d9517b9c948127319a09a7a36deac8')
>>> Web3.solidityKeccak(["bytes"], [Web3.toBytes(text=msg)])
HexBytes('0x1c8aff950685c2ed4bc3174f3472287b56d9517b9c948127319a09a7a36deac8')
>>> Web3.solidityKeccak(["bytes"], [Web3.toHex(text=msg)])
HexBytes('0x1c8aff950685c2ed4bc3174f3472287b56d9517b9c948127319a09a7a36deac8')
```